HeatStroke/GOComponents/GameObject.cpp

```diff
@@ -113,6 +120,66 @@ void GameObject::DeleteAllComponents()
 	m_mComponentMap.clear();
 }
 
+
+void GameObject::AddChild(GameObject* p_pChild)
+{
+	if (p_pChild != nullptr)
+	{
+		m_mChildMap.insert(std::pair<std::string, GameObject*>(p_pChild->GetGUID(), p_pChild));
+		p_pChild->SetParent(this);
+		m_Transform.AddChild(&(p_pChild->GetTransform()));
```

Comment:
Okay, well I'd need to sit down and have some debug prints and such to see this error through any further. For now I just wanted to point out the concern between those methods calling each other back and forth. I also see now that `SetParent()` is private. Have you checked out how it's done in HierarchicalTransform? It avoids methods calling each other and allows `SetParent()` to be public.